build_tools/python_deploy/promote_whl_from_rc_to_final.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This scripts takes a file like
+# 'iree_base_runtime-2.9.0rc20241107-cp311-cp311-manylinux_2_28_x86_64.whl'
+# with embedded version '2.9.0rc20241107' as input and then drops the
+# 'rcYYYYMMDD' suffix from both the embedded version and file name.
+#
+# Typical usage:
+#   pip install -r pypi_deploy_requirements.txt
+#   ./promote_whl_from_rc_to_final.py /path/to/file.whl --delete-old-wheel
+
+import argparse
+from change_wheel_version import change_wheel_version
+from packaging.version import Version
+from pathlib import Path
+from pkginfo import Wheel
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "input_file",
+        help="Path to the input .whl file to promote",
+        type=Path,
+    )
+    parser.add_argument(
+        "--delete-old-wheel",
```

Comment:
When you test locally, can you try with this flag set? By default the script keeps the original file around, which pip may be using (though your test command did try to install with an explicit file name 🤔)

Reply:
I think I did, but let me give it another try.

Reply:
Phew, thanks for checking. I considered having `promote_whl_from_rc_to_final.py` delete by default, but destructive operations can be frustrating to debug when iterating on scripts. The entry point `pypi_deploy.sh` script sets the flag explicitly.